Sources/RealModule/ApproximateEquality.swift

```diff
@@ -0,0 +1,154 @@
+  @inlinable @inline(__always)
+  public func isApproximatelyEqual(
+    to other: Self,
+    relativeTolerance: Magnitude = Magnitude.ulpOfOne.squareRoot()
+  ) -> Bool {
+    return isApproximatelyEqual(
+      to: other,
+      absoluteTolerance: relativeTolerance * Magnitude.leastNormalMagnitude,
+      relativeTolerance: relativeTolerance
+    )
+  }
+  
+  /// Compares `self` and `other` for approximate equality with a specified tolerances.
+  ///
+  /// `true` if `self` and `other` are equal, or if they are finite and either
+  /// ```
+  /// (self - other).magnitude ≤ absoluteTolerance
+  /// ```
+  /// or
+  /// ```
+  /// (self - other).magnitude ≤ relativeTolerance * scale
+  /// ```
+  /// where `scale` is `max(self.magnitude, other.magnitude)`.
+  ///
+  /// Mathematical Properties:
+  /// -
+  /// - `isApproximatelyEqual(to:absoluteTolerance:relativeTolerance:)`
+  ///   is _reflexive_ for non-exceptional values (such as NaN).
+  ///
+  /// - `isApproximatelyEqual(to:absoluteTolerance:relativeTolerance:)`
+  ///   is _symmetric_.
+  ///
+  /// - `isApproximatelyEqual(to:absoluteTolerance:relativeTolerance:)`
+  ///   is __not__ _transitive_. Because of this, approximately equality is
+  ///   __not an equivalence relation__, even when restricted to non-exceptional values.
+  ///
+  /// - For any point `a`, the set of values that compare approximately equal to `a` is _convex_.
+  ///   (Under the assumption that `norm` implements a valid norm, which cannot be checked
+  ///   by this function.)
+  ///
+  /// - Parameters:
+  ///
+  ///   - other: The value to which `self` is compared.
+  ///
+  ///   - absoluteTolerance: The absolute tolerance to use in the comparison.
+  ///
+  ///     This value should be non-negative and finite.
+  ///     This constraint on is only checked in debug builds, because a mathematically
+  ///     well-defined result exists for any tolerance, even one out of range.
+  ///
+  ///   - relativeTolerance: The relative tolerance to use in the comparison.
+  ///     Defaults to zero.
+  ///
+  ///     This value should be non-negative and less than or equal to 1.
+  ///     This constraint on is only checked in debug builds, because a mathematically
+  ///     well-defined result exists for any tolerance, even one out of range.
+  ///
+  ///   - norm: The norm to use for the comparison. Defaults to the `.magnitude`
+  ///     property.
+  ///
+  ///     When porting code or working with specialized algorithms, it may
+  ///     be desirable to override the norm. For example, if we wanted to test if
+  ///     a complex value was inside a circle of radius 0.001 centered at (1 + 0i), we
+  ///     could use:
+  ///     ```
+  ///     z.isApproximatelyEqual(
+  ///       to: 1,
+  ///       absoluteTolerance: 0.001,
+  ///       norm: \.length
+  ///     )
+  ///     ```
+  ///     (if we used the default, we would be testing if `z` were inside a square region
+  ///     instead.)
```

Comment:
This is a really helpful example! It'd be helpful if there were similar examples of using some of the other possible formulations (e.g. custom `relativeTolerance`, both an `absoluteTolerance` and a `relativeTolerance`).